libraries/lib-components/ModuleInterface.h

```diff
@@ -55,110 +56,132 @@ class PluginManagerInterface;
+   //! File types associated with this protocol
+   /*!
+    "Paths" returned by FindLibraryPaths() and passed back to
+    DiscoverPluginsAtPath() have provider-specific meaning. They are not
+    necessarily file system paths to existent files that could be placed in any
+    folder and queried for plug-in information.
+
+    This function returns nonempty only when that is the case, and lists the
+    possible extensions of such files (an empty string in a nonempty array means
+    any file is a candidate).
+    */
    virtual const FileExtensions &GetFileExtensions() = 0;
 
-   // Returns empty, or else, where to copy a plug-in file or bundle.
-   // Drag-and-drop is supported only if GetFileExtensions() returns nonempty and
-   // this function returns nonempty.
+   //! Where plug-in files should be copied to install them
+   /*!
+    @return may be empty. Drag-and-drop is supported only if GetFileExtensions()
+    returns nonempty and this function returns nonempty.
+    */
    virtual FilePath InstallPath() = 0;
 
-   // Modules providing a single or static set of plugins may use
-   // AutoRegisterPlugins() to register those plugins.
-   virtual bool AutoRegisterPlugins(PluginManagerInterface & pluginManager) = 0;
-
-   // For modules providing an interface to other dynamically loaded plugins,
-   // the module returns a list of path names that will be presented to the
-   // user as "New" for enablement.
-   virtual PluginPaths FindPluginPaths(PluginManagerInterface & pluginManager) = 0;
-
-   // Once the user selects desired paths from FindPluginPaths(),
-   // a call to DiscoverPluginsAtPath()
-   // will be made to request registration of one or more plugins.  If the module must create
-   // an instance of the plugin to register it, then the instance should be deleted
-   // after registration.
-   // May discover more than one plug-in at the path, and
-   // may call-back with paths not equal to path (perhaps appending
-   // other information to it).
-   // Error message does not need to mention the path and may be nonempty
-   // even if some plugins are also discovered successfully.
-   // Return value is the number of plugins found.
-   using RegistrationCallback =
-      std::function<
-         const PluginID &(ModuleInterface *, ComponentInterface *) >;
+   //! Called so that a provider of a static set of plug-ins can register them.
+   virtual void AutoRegisterPlugins(PluginManagerInterface & pluginManager) = 0;
+
+   //! Find available "paths", which may each be presented to the user, and
+   //! then reexamined (possibly loading libraries) to find one or more plug-ins
+   /*!
+    Paths are not necessarily file system paths.  Only the provider reinterprets
+    the paths.
+
+    Libraries may be associated with plug-ins, one-to-many.
+    @see GetFileExtensions DiscoverPluginsAtPath
+    */
+   virtual PluginPaths
+      FindLibraryPaths(PluginManagerInterface & pluginManager) = 0;
```

Comment:
I was trying to avoid the term "module" because I thought it was imprecise terminology, and much other un-merged work of mine is going to use the term for feature add-ons that are not necessarily related to effects.

You want to change a function name, not the type alias `PluginPaths`?

Reply:
I can agree to use the term "module" generally to mean some dynamically loaded library in this sense, as well as for an add-on.  But the old use of "module" did not refer to that and I would rather use the term "provider" which was already present in some comments and function names.

Reply:
If "library" does not mean "system library", then I don't have objections